s3buildcache/src/test/kotlin/androidx/build/gradle/s3buildcache/S3StorageServiceTest.kt

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package androidx.build.gradle.s3buildcache
+
+import org.junit.Test
+
+/**
+ * This test needs to be configured with the correct values for REGION, BUCKET_NAME,
+ * and AWS credentials, therefore, it won't pass on CI as is.
+ * */
+class S3StorageServiceTest {
```

Comment:
Do we really want to add a test that doesnt run? It would be a lot nicer if we could have some sort of local implementation backing it.

Reply:
Yea good point. Updated tests to be backed by a [mock web server](https://github.com/adobe/S3Mock)